src/kernel/src/memory/context/virtmem.rs

```diff
@@ -650,30 +714,34 @@ pub fn page_fault(addr: VirtAddr, cause: MemoryAccessKind, flags: PageFaultFlags
             if let Some((page, cow)) =
                 obj_page_tree.get_page(page_number, cause == MemoryAccessKind::Write)
             {
-                ctx.arch.map(
-                    info.mapping_cursor(page_number.as_byte_offset(), PageNumber::PAGE_SIZE),
-                    &mut info.phys_provider(&page),
-                    &info.mapping_settings(cow, is_kern_obj),
-                );
-                ctx.arch.change(
-                    info.mapping_cursor(page_number.as_byte_offset(), PageNumber::PAGE_SIZE),
-                    &info.mapping_settings(cow, is_kern_obj),
-                );
+                ctx.with_arch(KERNEL_SCTX, |arch| {
```

Comment:
Maybe add a note here so we do not forget to change this when we have support for user security contexts.